odoo/addons/base/models/ir_http.py

```diff
@@ -325,6 +325,11 @@ def _pre_dispatch(cls, rule, args):
 
     @classmethod
     def _dispatch(cls, endpoint):
+        # Verify the captcha in case it was set on @http.route
+        captcha = endpoint.routing.get('captcha')
+        if captcha:
+            if request.httprequest.method == 'POST' and not request.env['ir.http']._verify_request_recaptcha_token(captcha):
```

Comment:
I care about line-length in the modules I maintain :p
```suggestion
        if (
            captcha := endpoint.routing.get('captcha')
            and request.httprequest.method == 'POST'
            and not request.env['ir.http']._verify_request_recaptcha_token(captcha)
        ):
```
Btw what's about the other http methods? Like `PUT` and `DELETE`. Souldn't we also check the captcha for them? Shouldn't we actually *skip* the captcha verification for [safe](https://httpwg.org/specs/rfc9110.html#safe.methods) methods?

```suggestion
        # https://httpwg.org/specs/rfc9110.html#safe.methods
        SAFE_HTTP_METHODS = ('GET', 'HEAD', 'OPTION', 'TRACE')  # move me in http.py
        if (
            captcha := endpoint.routing.get('captcha')
            and request.httprequest.method not in SAFE_HTTP_METHODS
            and not request.env['ir.http']._verify_request_recaptcha_token(captcha)
        ):
```